lib/grape/middleware/stack.rb

```diff
@@ -90,8 +91,13 @@ def merge_with(middleware_specs)
       def build(builder = Rack::Builder.new)
         others.shift(others.size).each(&method(:merge_with))
         middlewares.each do |m|
-          opts = m.args.extract_options!
-          m.block ? builder.use(m.klass, *m.args, **opts, &m.block) : builder.use(m.klass, *m.args, **opts)
+          if RUBY_VERSION >= '2.7'
```

Comment:
This won't work after version 10.

```ruby
2.6.6 :001 > RUBY_VERSION 
 => "2.6.6" 
2.6.6 :002 > RUBY_VERSION > '2.7.7'
 => false 
2.6.6 :003 > RUBY_VERSION > '2.6.1'
 => true 
2.6.6 :004 > RUBY_VERSION > '16.1'
 => true 
```

Lets find a way to avoid this `if`?

Reply:
You could use https://github.com/dblock/with-version, but I am not sure we want a dependency for syntax sugar, so use something like https://metaredux.com/posts/2018/10/28/a-better-way-to-compare-versions-in-ruby.html if you must.

If we don't have another way, take the `if` outside of the execution part, and declare separate method implementations for different versions. This way we don't re-evaluate the same thing again and again during execution.

Reply:
Apologies, fixed this a while ago but forgot to push :facepalm: